src/LinuxCaptureService/LinuxCaptureService.cpp

```diff
@@ -227,9 +223,21 @@ grpc::Status LinuxCaptureService::Capture(
     grpc::ServerReaderWriter<CaptureResponse, CaptureRequest>* reader_writer) {
   orbit_base::SetCurrentThreadName("CSImpl::Capture");
 
-  if (grpc::Status result = InitializeCapture(reader_writer); !result.ok()) {
-    return result;
+  orbit_capture_service::ClientCaptureEventCollectorBuilderImpl
+      client_capture_event_collector_builder{reader_writer};
+
+  // shared_ptr because it might outlive this method. See wait_for_stop_capture_request_thread_ in
+  // WaitForStopCaptureRequestOrMemoryThresholdExceeded.
+  auto start_stop_capture_request_waiter =
+      std::make_shared<orbit_capture_service::StartStopCaptureRequestWaiterImpl>(reader_writer);
+
+  if (CaptureServiceBase::CaptureInitializationResult result =
+          InitializeCapture(&client_capture_event_collector_builder);
+      result == CaptureServiceBase::CaptureInitializationResult::kAlreadyInProgress) {
+    return {grpc::StatusCode::ALREADY_EXISTS,
+            "Cannot start capture because another capture is already in progress"};
   }
+
```

Comment:
I'd add a CHECK here, that the status code is actually SUCCESS, in case we add other error types in the future.

Reply:
A way to protect us against this would be to have a `switch` on `result` instead, with no `default`.

Reply:
Would also prefer the switch.

Reply:
Changed to use `switch`. Thanks for the suggestion!